chore: rewrite documents and split out sections (#117)
IBM/cloudant-kafka-connector: docs/smt-reference.md

    ```
    transforms=RenameField
    transforms.RenameField.type=org.apache.kafka.connect.transforms.ReplaceField$Value 
    transforms.RenameField.renames=name:_id
    ```
1. If you have `_id` fields and would prefer to have Cloudant generate a UUID for the document ID, use the `ReplaceField` transform to exclude the existing `_id` field:
    ```
    transforms=ReplaceField
    transforms.ReplaceField.type=org.apache.kafka.connect.transforms.ReplaceField$Value 
    transforms.ReplaceField.exclude=_id
    ```
1. If you have events where there is no value ([_tombstone_ events](https://kafka.apache.org/documentation.html#compaction)), you may wish to filter these out.
    - In the Cloudant sink connector, these may be undesirable as they will generate an empty document.
    - In the Cloudant source connector, tombstone events are generated for deleted documents (in addition to the deleted document itself).
    - In either case, you can use the `RecordIsTombstone` predicate with a filter to remove these tombstone events as shown in this example:

    ```

>     transforms=dropNullRecords
>     transforms.dropNullRecords.type=org.apache.kafka.connect.transforms.Filter
>     transforms.dropNullRecords.predicate=isNullRecord
> 
>     predicates=isNullRecord
>     predicates.isNullRecord.type=org.apache.kafka.connect.transforms.predicates.RecordIsTombstone

Reviewer: We could reduce the naming discrepencies here a little further
```suggestion
    transforms=dropNullEvents
    transforms.dropNullEvents.type=org.apache.kafka.connect.transforms.Filter
    transforms.dropNullEvents.predicate=isNullEvent

    predicates=isNullEvent
    predicates.isNullEvent.type=org.apache.kafka.connect.transforms.predicates.RecordIsTombstone
```

   the `docid` event key to the `cloudant_doc_id` custom header and sets the header converter to string:
   ```
   transforms=moveFieldsToHeaders
   transforms.moveFieldsToHeaders.type=org.apache.kafka.connect.transforms.HeaderFrom$Key
   transforms.moveFieldsToHeaders.fields=docid
   transforms.moveFieldsToHeaders.headers=cloudant_doc_id
   transforms.moveFieldsToHeaders.operation=move
   
   header.converter=org.apache.kafka.connect.storage.StringConverter
   ```

   **Note**: The `header.converter` is required to be set to `StringConverter` since the document ID field only supports strings.

1. If you have events where the `_id` field is absent or `null` then Cloudant will generate
   a document ID. If you don't want this to happen then set an `_id` (see earlier examples).
   If you need to filter out those documents or drop `_id` fields when the value is `null` then you'll need to create a custom SMT.

**Note**: For any of the SMTs above, if the field does not exist it will leave the event unmodified and continue processing the next event.

## Source

> The examples below demonstrate modifying records produced by the Cloudant source connector.

Reviewer: ```suggestion
The examples below demonstrate modifying events produced by the Cloudant source connector.
```

1. Flatten maps in the JSON document using the Kafka built-in `org.apache.kafka.connect.transforms.ReplaceField$Value`
    ```
    transforms=FlattenMaps
    transforms.FlattenMaps.type=org.apache.kafka.connect.transforms.Flatten
    ```

1. Flatten arrays in the JSON document using `com.ibm.cloud.cloudant.kafka.connect.transforms.ArrayFlatten`. Note that this transform

>    is only suitable for use with Map record values and will filter records that do not conform. As such if used in conjunction with the

Reviewer: ```suggestion
   is only suitable for use with Map event values and will filter events that do not conform. As such if used in conjunction with the
```

   `MapToStruct` transform, this `ArrayFlatten` operation must precede `MapToStruct` in the SMT pipeline.
   The `delimiter` configuration property may be used to customize the delimiter, which defaults to `.`.
    ```
    transforms=FlattenArrays
    transforms.FlattenArrays.type=com.ibm.cloud.cloudant.kafka.connect.transforms.ArrayFlatten
    ```

1. Convert schemaless `java.util.Map` values to `org.apache.kafka.connect.data.Struct` with an inferred schema. This transform is designed

>    to improve compatibility with other connectors and converters that requires a `Struct` type record. For complex schemas a schema registry

Reviewer: ```suggestion
   to improve compatibility with other connectors and converters that requires a `Struct` type event. For complex schemas a schema registry
```
